WcaOnRails/app/controllers/api/v0/user_roles_controller.rb

```diff
@@ -191,11 +191,28 @@ def index_for_group_type
     render json: roles
   end
 
+  private def end_delegate_role_for_user(user)
+    if user.delegate_status.present?
+      remove_pending_wca_id_claims(user)
+      user.update!(delegate_status: '', region_id: '', location: '')
+      send_role_change_notification(user)
+    end
+  end
+
+  private def end_role_for_user_in_group_with_status(group, status)
+    if group.group_type == UserGroup.group_types[:delegate_regions]
+      user = User.find_by(region_id: group.id, delegate_status: status)
+      if user.present?
+        end_delegate_role_for_user(user)
+      end
+    end
+  end
+
   def create
     user_id = params.require(:userId)
     group_id = params.require(:groupId)
 
-    if group_id.include?("_") # Temporary hack to support some old system roles, will be removed once all roles are
+    if group_id.is_a?(String) && group_id.include?("_") # Temporary hack to support some old system roles, will be removed once all roles are
       # migrated to the new system.
       group_type = group_id.split("_").first
       original_group_id = group_id.split("_").last
```

Comment:
Maybe this needs the "drop reverse drop last reverse reverse first backwards" hack from the other PR now?

Reply:
Actually yes & no. Yes for consistency, hence I've added it now. No because this will be used only by group_type councils which doesn't have `_`. But maybe we will have it in future, so good to have.